### PR TITLE
modify writeData to write complete interface variable outputs in netCDF

### PR DESCRIPTION
**Pull request by luckymichael**
interface variables have dimension (nLayer + 1).

### CODE REVIEW

File: build/source/netcdf/modelwrite.f90 [hunk 1]
```diff
@@ -188,19 +188,19 @@ subroutine writeData(modelTimestep,outputTimestep,meta,stat,dat,map,indx,iHRU,er
...
          case(iLookVarType%midSoil); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,midSoilStartIndex/),count=(/1,nSoil/))
-         case(iLookVarType%ifcToto); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcTotoStartIndex/),count=(/1,nLayers/))
-         case(iLookVarType%ifcSnow); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSnowStartIndex/),count=(/1,nSnow/))
-         case(iLookVarType%ifcSoil); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSoilStartIndex/),count=(/1,nSoil/))
+         case(iLookVarType%ifcToto); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcTotoStartIndex/),count=(/1,nLayers+1/))
+         case(iLookVarType%ifcSnow); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSnowStartIndex/),count=(/1,nSnow+1/))
+         case(iLookVarType%ifcSoil); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSoilStartIndex/),count=(/1,nSoil+1/))
```
Comment (bartnijssen) [inline build/source/netcdf/modelwrite.f90]: @luckymichael : Does the `+1` array element always exist?

**Comment by luckymichael** [comment]
yes. because they are the interfaces of layers. @martynpclark could you confirm this?

**Comment by bartnijssen** [comment]
@martynpclark : Can you respond to Michael's query? if everything looks OK I can go ahead and merger this.

**Comment by bartnijssen** [comment]
@martynpclark : Please see question above.

@luckymichael : Please resolve conflicts and update the PR

**Comment by luckymichael** [comment]
@bartnijssen: conflicts resolved.

**Pull request merged**